### PR TITLE
fix: Add all Trino dependencies for Kerberos enablement

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

Per our [documentation](https://superset.apache.org/docs/databases/trino/#2-kerberos-authentication), Superset supports Kerberos authentication for Trino, however installing the `superset[trino]` dependencies are not suffice as the `trino` package additionally requires the [requests-kerberos](https://github.com/trinodb/trino-python-client/blob/f99ef245f28b4e1784a829b7935a31d0215ace4d/setup.py#L29-L34) package.

Rather than defining `trino[kerberos]` as a dependency I though it would be prudent to defining `trino[all]` as this contains additional requirements like SQLAlchemy.

### TESTING INSTRUCTIONS

Ran `pip-compile-multi --no-upgrade`.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -371,6 +369,7 @@ werkzeug==2.3.3
     # via
     #   apache-superset
     #   flask
+    #   flask-appbuilder
```
Comment (villebro) [inline requirements/base.txt:372]: I think we need to add a CI linting rule to make sure the requirements files are always up to date (I assume this change didn't bring about this change). Any objections to me adding a step to the python lint CI workflow that does this in a separate PR?

**Review by villebro — APPROVED** [review]
LGTM

File: requirements/base.txt (hunk 1 above)
Comment (john-bodley) [reply requirements/base.txt:372]: @villebro I agree. We do have [this](https://github.com/apache/superset/blob/master/.pre-commit-config.yaml#L59-L62) check, however this just makes sure that the SHAs are correct. Maybe we need to run `pip-compile-multi --no-upgrade` and make sure there's no resulting diff?

Comment (villebro) [reply requirements/base.txt:372]: > Maybe we need to run pip-compile-multi --no-upgrade and make sure there's no resulting diff?

@john-bodley yeah, I was thinking something along those lines

**Comment by john-bodley** [comment]
Closing in favor of https://github.com/apache/superset/pull/26119 as we likely don't want to bloat our CI dependencies when there is an alternative workaround.

**Pull request closed without merging**